### PR TITLE
fix: load script packages before loading builtin assets

### PR DESCRIPTION
**Pull request by PPpro**
Re: https://github.com/cocos/3d-tasks/issues/17747#issuecomment-1735286855

### NOTE

now, our internal bundle has script deps, so we need to load script packages first

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
What happens if the script refer to some builtin assets?

**Comment by PPpro** [comment]
> What happens if the script refer to some builtin assets?

it not because the script refer to some builtin assets, it is because internal bundle includes some scripts which rely on `scriptsPackages`

so we need to load `scriptsPackages` before we load builtin assets, the builtin assets includes the internal bundle

**Review by minggo — APPROVED** [review]

File: cocos/game/game.ts [hunk 1]
```diff
@@ -795,27 +799,36 @@ export class Game extends EventTarget {
                 }
                 const data = effectSettings.data;
                 if (data === null) {
-                    console.error('Effect settings not found, effects will not be imported.');
+                    error('Effect settings not found, effects will not be imported.');
```
Comment (pandamicro) [inline cocos/game/game.ts]: Please change to errorID

File: cocos/game/game.ts [hunk 2]
```diff
@@ -1013,11 +1021,11 @@ export class Game extends EventTarget {
             SplashScreen.instance.update(this._calculateDT(false));
         } else if (this._shouldLoadLaunchScene) {
             this._shouldLoadLaunchScene = false;
-            const launchScene = settings.querySettings(Settings.Category.LAUNCH, 'launchScene');
+            const launchScene = settings.querySettings(Settings.Category.LAUNCH, 'launchScene') as string;
             if (launchScene) {
                 // load scene
                 director.loadScene(launchScene, (): void => {
-                    console.log(`Success to load scene: ${launchScene}`);
+                    log(`Success to load scene: ${launchScene}`);
```
Comment (pandamicro) [inline cocos/game/game.ts]: logID here

**Pull request merged**